### DISCUSSION TITLE
Allowing products to have varying "Last date of modifications"

### DISCUSSION THREAD

**Discussion by theschweitz** [pretix/pretix#2955] [Ideas]
Currently, Pretix only has one overall setting for "Last Date of Modifications" that applies to every product. However, there are many cases where certain products should be set-in-stone, and others should be editable.

Right now, there is the option for "Allow products to be canceled or changed," however, we don't necessarily want folks to be able to **cancel** their order AND change the details as one option. Because of our plugin that allows users to pay for tickets with crypto-payments, we require that folks reach out to us to request refunds, so having the "Cancel" option available isn't as desirable as the "Last date of modifications" option.

For example: We want attendees who order **General Admission** tickets to be able to change the email address on their ticket in the event they can no longer attend & wish to transfer the ticket to a friend.
**However**, we grant Discounted Tickets to those who have been accepted via an external application (Google Form), and once these users fill in their order information, we do not want them to be able to change the data on the ticket without our permission, (e.g. in the event that they wish to send their ticket to a friend who did not qualify for the Discount).

We understand the task of adding these types of settings to each product; But — it can be as easy as making a _General_  "Last Date of Modifications" that applies to _every product_, and adding a  "Last Date of Modifications" to _each product's settings_ & making this selection on the specific product override the General.

Eager to brainstorm on this!

**Comment by raphaelm** [comment]
Yeah good point. We had a previous feature request to block modifications for all tickets that went through an approval process, but I think your idea of making it a product option is the better one. So yes, I think we do want this feature.

The main open question here is what happens with mixed orders. Currently, if just one product in an order does not allow e.g. cancelling or changing its parameters, changing the order is blocked entirely. We'd need to decide whether this is a good-enougj approach here as well, or whether we need people to be able to partially modify orders, which would be more powerful but also harder to implement.
